### PR TITLE
update #945 - add hamburger menu on mobile devices

### PR DESCRIPTION
**Pull request by Ulisseus**
Current navBar functionality is rather limited on phones and tablets, you can only log in or sign up. Which makes sense considering that there isn't much space on smaller screens.

This PR adds hamburger menu which makes it possible to have nice mobile layout while keeping full functionality from desktop component.

### AppNav

I switched from using bootsrap to react-boostrap for navBar. Normal bootsrap is using jquery for updating the dom (which is obviously bad for react project), so there is no bootrap js in our bundle and more complex components like navBar don't work. 

* Username is now capitalized.
* Login and Logout buttons have the same width. 
* Admin dropdown uses the same styling as other buttons.
* Small fix to route highlighting. It still doesn't work with admin components but it's a feature at this point.
* Bazillion updated snapshots 

[Vercel](https://c0d3-app-nr3h0rk2s-c0d3.vercel.app).

### CODE REVIEW

File: components/DropdownMenu.tsx [hunk 1]
```diff
@@ -23,33 +24,49 @@ type DropDownMenuProps = {
     | 'warning'
     | 'danger'
     | 'none'
+  bsPrefix?: string
```
Comment (hwong0305) [inline components/DropdownMenu.tsx:27]: what is `bsPrefix`? Can you please give this a better name?

File: scss/appNav.module.scss [hunk 2]
```diff
@@ -7,3 +7,21 @@
...
+  & > * {
+    min-width: 5em;
+    @media (min-width: 992px) {
+      background-color: white !important;
+    }
+  }
+}
+
+.dropdown-item {
+  color: #838383;
```
Comment (hwong0305) [inline scss/appNav.module.scss:22]: This color `#838383` is used really often. Should this be setup in our theme instead?

**Review by songz — APPROVED** [review]

File: components/DropdownMenu.tsx (hunk 1 above)
Comment (Ulisseus) [reply components/DropdownMenu.tsx:29]: I added comment describing how bsPrefix works in bootsrap.

File: scss/appNav.module.scss (hunk 2 above)
Comment (Ulisseus) [reply scss/appNav.module.scss:22]: I created new file colors.module.scss from which we can import default colors. It wasn't possible to import from index (it uses global styles). 

TO-DO. Change all colors in scss files to import from this new file.

**Pull request merged**